datadog_checks_base/datadog_checks/base/checks/openmetrics/mixins.py

```diff
@@ -627,3 +650,14 @@ def _metric_tags(self, metric_name, val, sample, scraper_config, hostname=None):
 
     def _is_value_valid(self, val):
         return not (isnan(val) or isinf(val))
+
+    def _get_bearer_token(self, bearer_token_auth, bearer_token_path):
+        if bearer_token_auth is False:
+            return None
+        path = bearer_token_path or self.DEFAULT_BEARER_TOKEN_PATH
```

Comment:
maybe instead of defaulting to the Kubernetes serviceaccount token. the check should do an auto-discovery of the file. 

```python
path == None
if os.path.isfile(bearer_token_path):
    path = bearer_token_path
else if os.path.isfile(self.KUBERNETES_TOKEN_PATH): 
    path = self.KUBERNETES_TOKEN_PATH
else if os.path.isfile(self.DOCKER_SWARM_TOKEN_PATH): 
    path = self.DOCKER_SWARM_TOKEN_PATH
if path == None:
    self.log.error("Cannot get bearer token from "bearer_token_path" or auto discovery: {} - error: {}".format(path, err))
```

Reply:
I agree it's better this way  👍 although we may need to exclude auto discovering in docker swarm since they don't mount a file containing the bearer token apparently

Reply:
`DOCKER_SWARM_TOKEN_PATH` was only an example. I'm sure in the future, other solution will come :)